### PR TITLE
Update whitelist

### PR DESCRIPTION
**Pull request by no-boot-device**
* Remove defunct pomf clones (aka all of them, god rest their souls): This is important for safety, as if someone were to buy up their domain they could use it for EEEVIL
* Add IPFS, a distributed file/website/etc host
* Add neocities, a geocities replacement and therefore a filehost (if I fucked up the regex pls lemme know and/or fix it for my lazy ass)

### CODE REVIEW

**Comment by thegrb93** [comment]
I just need to check neocities' services to see what kind of things it allows

**Comment by thegrb93** [comment]
Neocities looks fine, but I don't think that's a valid pattern https://www.lua.org/manual/5.3/manual.html#6.4.1

File: lua/starfall/permissions/providers_sh/url_whitelist.lua [hunk 1]
```diff
@@ -180,3 +167,15 @@ simple [[i.redd.it]]
...
+--- Examples:
+--- https://ipfs.io/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/I/m/Ellis_Sigil.jpg
+
+simple [[ipfs.io]]
+
+-- neocities
+--- Examples:
+--- https://fauux.neocities.org/LainDressSlow.gif
+
+pattern [[[A-Za-z0-9](?:[A-Za-z0-9%-]{0,61}[A-Za-z0-9])?%.neocities%.org/(.+)]
```
Comment (Damianu) [inline lua/starfall/permissions/providers_sh/url_whitelist.lua]: pattern [[([%w-_]+)%.neocities%.org/(.+)]]
Seems a lot better, also you forgot ] at the end

Comment (thegrb93) [reply lua/starfall/permissions/providers_sh/url_whitelist.lua]: Will that defend against blah.neocities.org.mywebsite.com?

Comment (thegrb93) [reply lua/starfall/permissions/providers_sh/url_whitelist.lua]: Ah just noticed the `/` maybe that will prevent that

Comment (Damianu) [reply lua/starfall/permissions/providers_sh/url_whitelist.lua]: It's modified discord pattern, I tested them all so I am sure its fine :)

Comment (thegrb93) [reply lua/starfall/permissions/providers_sh/url_whitelist.lua]: All right, nice.

**Pull request merged**